pkgs/build-support/fetchgit/nix-prefetch-git

```diff
@@ -348,6 +356,23 @@ clone_user_rev() {
     fi
 }
 
+clone_user_rev_to_tmpfile(){
+    local url="$1"
+    local rev="${2:-HEAD}"
+
+    # nix>=2.20 rejects adding symlinked paths to the store, so use realpath
+    # to resolve to a physical path. https://github.com/NixOS/nix/issues/11941
+    tmpPath="$(realpath "$(mktemp -d --tmpdir git-checkout-tmp-XXXXXXXX)")"
+    exit_handlers+=(remove_tmpPath)
+
+    tmpOut="$tmpPath/$storePathName"
+    tmpClone="$tmpPath/clone"
```

Comment:
I think it's possible that `$storePathName` could be `clone`, in which case these two values will be the same. That will cause a failure later when running the `mv "$tmpClone" "$tmpOut"` or `mv "$tmpClone/$rootDir" "$tmpOut"` commands.

I'd suggest either (a) using `mktemp` to create two separate temporary directories, or (b) creating `"$tmpPath/out"` and setting `tmpOut` to be `$tmpPath/out/$storePathName`.